### DISCUSSION TITLE
about upload

### DISCUSSION THREAD

**Discussion by w1249841003** [zauberzeug/nicegui#1140] [Q&A] [answered]
### Question

I need to use the click button to select files and upload files, but I found that upload only has reset and delete. I found upload.py in the source code, and created a new function pickFiles with reference to reset, which can be used normally.
I would like to ask, how to use these functions without modifying the source code?

**Comment by w1249841003** [comment]
```py
from nicegui import ui, events

def upload(e: events.UploadEventArguments):
    with open(e.name, 'wb') as f:
        f.write(e.content.read())

def button_click():
    """get e.name"""
    pass

file_upload = ui.upload(on_upload=upload, auto_upload=True)
button_submit = ui.button('Submit', on_click=button_click)
ui.run(native=True)
```


There is another question, how to call the uploaded file more conveniently? It can be passed to the function to save the file through on_upload, but I need to click the button to process it.
I can use upload_file.name in streamlit, is there a similar method in nicegui?

**Comment by rodja** [comment]
I'm sorry @w1249841003. I do not understand what you want to archive. What do you mean with

> I need to use the click button to select files and upload files

Do you want to auto-upload as soon as a file is selected? That can be archived with the [Quasar prop `auto-upload`](https://quasar.dev/vue-components/uploader#introduction):

```py
ui.upload(on_upload=lambda e: ui.notify(f'Uploaded {e.name}')).props('auto-upload')
```

**Reply by w1249841003** [reply]
![1688702560365](https://github.com/zauberzeug/nicegui/assets/25957813/a5e45482-c674-461e-acb8-91eac1da6367)
Sorry, I translated it through google.
For example, click a button to select a file or start uploading. I saw pictures with this method.
Currently this is how I do it: add the following function in upload.py
```
     def pickFiles(self) -> None:
         self. run_method('pickFiles')
```
Then in my code add
```
def button_click():
     file_upload. pickFiles()
```
In this way, the file selection window can be popped up after clicking the button

**Reply by w1249841003** [reply]
I need to achieve: This functionality can be implemented without modifying [upload.py]

**Reply by falkoschindler** [reply, marked as answer]
Like this?
```py
def button_click():
     file_upload.run_method('pickFiles')
```

**Reply by w1249841003** [reply]
yes, very good thank you
